File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleNativeAd.java
package com.vungle.mediation;

import android.content.Context;
import android.util.Log;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.ads.mediation.vungle.VungleMediationAdapter;
import com.vungle.ads.AdConfig;
import com.vungle.ads.BaseAdListener;
import com.vungle.ads.NativeAd;
import com.vungle.ads.NativeAdLayout;
import com.vungle.ads.internal.ui.view.MediaView;

/**
 * This class is used to represent a Vungle Native ad.
 */
public class VungleNativeAd {

    private final String placementId;

    private final NativeAdLayout nativeAdLayout;

    private final MediaView mediaView;
    private final NativeAd nativeAd;

    /**
     * Vungle ad object for native ads.
     */
    public VungleNativeAd(
            @NonNull Context context,
            @NonNull String placementId,
            @NonNull AdConfig adConfig,
            boolean isLifeCycleManagementDisabled,
            @NonNull BaseAdListener baseAdListener) {
        this.placementId = placementId;
        this.nativeAd = new NativeAd(placementId, adConfig);
        nativeAd.setAdListener(baseAdListener);
        this.nativeAdLayout = new NativeAdLayout(context);
        this.nativeAdLayout.disableLifeCycleManagement(isLifeCycleManagementDisabled);
        this.mediaView = new MediaView(context);
    }

    public void loadNativeAd(@Nullable String adMarkup) {
        nativeAd.load(adMarkup);
    }

    @Nullable
    public NativeAd getNativeAd() {
        return nativeAd;
    }

    public NativeAdLayout getNativeAdLayout() {
        return nativeAdLayout;
    }

    public MediaView getMediaView() {
        return mediaView;
    }

    // TODO: this method isn't being called
    public void destroyAd() {
        if (nativeAdLayout != null) {
            nativeAdLayout.removeAllViews();
            if (nativeAdLayout.getParent() != null) {
                ((ViewGroup) nativeAdLayout.getParent()).removeView(nativeAdLayout);
            }
        }

        if (mediaView != null) {
            mediaView.removeAllViews();
            if (mediaView.getParent() != null) {
                ((ViewGroup) mediaView.getParent()).removeView(mediaView);
            }
        }

        if (nativeAd != null) {
            Log.d(VungleMediationAdapter.TAG, "Vungle native adapter cleanUp: destroyAd # " + nativeAd.hashCode());
            nativeAd.unregisterView();
            nativeAd.destroy();
        }
    }

    @NonNull
    @Override
    public String toString() {
        return " [placementId="
                + placementId
                + " # nativeAdLayout="
                + nativeAdLayout
                + " # mediaView="
                + mediaView
                + " # nativeAd="
                + nativeAd
                + " # hashcode="
                + hashCode()
                + "] ";
    }
}

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleNativeAd.kt
package com.vungle.mediation

import android.content.Context

/**
 * This class is used to represent a Vungle Native ad.
 */
class VungleNativeAd(
    context: Context,
    private val placementId: String,
    isLifeCycleManagementDisabled: Boolean
) {
//    val nativeAdLayout: NativeAdLayout?
//    val mediaView: MediaView?
//
//    /**
//     * Vungle ad object for native ads.
//     */
//    private val nativeAd: NativeAd?
//    fun loadNativeAd(adConfig: AdConfig?, adMarkup: String?, listener: NativeAdListener?) {
//        nativeAd.loadAd(adConfig, adMarkup, listener)
//    }
//
//    fun getNativeAd(): NativeAd? {
//        return nativeAd
//    }
//
//    fun destroyAd() {
//        if (nativeAdLayout != null) {
//            nativeAdLayout.removeAllViews()
//            if (nativeAdLayout.parent != null) {
//                (nativeAdLayout.parent as ViewGroup).removeView(nativeAdLayout)
//            }
//        }
//        if (mediaView != null) {
//            mediaView.removeAllViews()
//            if (mediaView.parent != null) {
//                (mediaView.parent as ViewGroup).removeView(mediaView)
//            }
//        }
//        if (nativeAd != null) {
//            Log.d(
//                VungleMediationAdapter.TAG,
//                "Vungle native adapter cleanUp: destroyAd # " + nativeAd.hashCode()
//            )
//            nativeAd.unregisterView()
//            nativeAd.destroy()
//        }
//    }
//
//    override fun toString(): String {
//        return (" [placementId="
//                + placementId
//                + " # nativeAdLayout="
//                + nativeAdLayout
//                + " # mediaView="
//                + mediaView
//                + " # nativeAd="
//                + nativeAd
//                + " # hashcode="
//                + hashCode()
//                + "] ")
//    }
//
//    init {
//        nativeAd = NativeAd(context, placementId)
//        nativeAdLayout = NativeAdLayout(context)
//        nativeAdLayout.disableLifeCycleManagement(isLifeCycleManagementDisabled)
//        mediaView = MediaView(context)
//    }
}